File: backend/internal/domain/entity/portfolio.go
package entity

type PortfolioOverview struct {
	TotalProjects     int       `json:"total_projects"`
	ActiveProjects    int       `json:"active_projects"`
	CompletedProjects int       `json:"completed_projects"`
	OnHoldProjects    int       `json:"on_hold_projects"`
	TotalBudget       float64   `json:"total_budget"`
	GreenProjects     int       `json:"green_projects"`
	YellowProjects    int       `json:"yellow_projects"`
	RedProjects       int       `json:"red_projects"`
	HighRiskProjects  []Project `json:"high_risk_projects"`
}

File: frontend/src/lib/services/portfolio.ts
import { apiFetch } from './api';

const API_URL = '/api/v1';

export interface PortfolioProject {
    id: number;
    project_id: string;
    project_name: string;
    project_status: string;
    overall_health: string;
    progress: number;
    approved_budget: number;
    actual_cost: number;
    planned_end_date?: string | null;
}

export interface PortfolioOverview {
    total_projects: number;
    active_projects: number;
    completed_projects: number;
    on_hold_projects: number;
    total_budget: number;
    green_projects: number;
    yellow_projects: number;
    red_projects: number;
    high_risk_projects: PortfolioProject[];
}

export const portfolioService = {
    async getOverview(): Promise<PortfolioOverview> {
        const res = await apiFetch(`${API_URL}/portfolio/overview`);
        if (!res.ok) throw new Error('Failed to load portfolio data');
        return res.json();
    }
};
